build_torcx_store

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+# Copyright (c) 2017 The CoreOS Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+. "$(dirname "$0")/common.sh" || exit 1
+
+# Script must run inside the chroot
+restart_in_chroot_if_needed "$@"
+
+assert_not_root_user
+
+# Developer-visible flags.
+DEFINE_string board "${DEFAULT_BOARD}" \
+  "The board to build packages for."
+DEFINE_string output_root "${DEFAULT_BUILD_ROOT}/torcx" \
+  "Directory in which to place torcx stores (named by board/version)"
+
+# include upload options
+. "${BUILD_LIBRARY_DIR}/release_util.sh" || exit 1
+
+FLAGS_HELP="usage: $(basename $0) [flags] [images]
+
+This script builds a collection torcx images in the given directory that can be
+used as a torcx store directly.  By default, all supported images are built,
+but a list of images can be given as command arguments.  Note that their order
+matters, since the version specified last will get the default ref symlink.
+"
+show_help_if_requested "$@"
+
+# The following options are advanced options, only available to those willing
+# to read the source code. They are not shown in help output, since they are
+# not needed for the typical developer workflow.
+DEFINE_integer build_attempt 1 \
+  "The build attempt for this image build."
+DEFINE_string group developer \
+  "The update group."
+DEFINE_string version '' \
+  "Overrides version number in name to this version."
+
+# Parse command line
+FLAGS "$@" || exit 1
+eval set -- "${FLAGS_ARGV}"
+
+# Only now can we die on error.  shflags functions leak non-zero error codes,
+# so will die prematurely if 'switch_to_strict_mode' is specified before now.
+switch_to_strict_mode
+
+# Define BUILD_DIR and set_build_symlinks.
+. "${BUILD_LIBRARY_DIR}/toolchain_util.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/board_options.sh" || exit 1
+. "${BUILD_LIBRARY_DIR}/build_image_util.sh" || exit 1
+
+# Print the first level of runtime dependencies for a torcx meta-package.
+function torcx_dependencies() (
+        local pkg=${1:?}
+        local ebuild inherit
+        ebuild=$(equery-${BOARD} w "${pkg}")
+        function inherit() { : ; }
+        . "${ebuild}"
+        echo ${RDEPEND}
+)
+
+# Build and install a package configured as part of a torcx image.
+function torcx_build() (
+        local pkg=${2:?}
+        local tmproot=${1:?}
+
+        export LDFLAGS=-Wl,-rpath,/ORIGIN/../lib
+        export PKGDIR="${tmproot}/var/lib/portage/pkgs"
+
+        # Allow the meta-package to install bashrc to customize the builds.
```

Comment:
Oh, and there isn't a reason for it at the moment with only Docker packages.  But we don't have the equivalent of real portage profiles here to support `/etc/portage/bashrc` since we're using the regular board profiles.  We can add a subprofile for this that inherits the board profile, but that will increase complexity again.  The idea behind a `bashrc` here is to support different `INSTALL_MASK`s etc. that must differ for packages between torcx and prod boards.